packages/oas-to-snippet/src/index.ts

```diff
@@ -68,21 +88,18 @@ export default async function oasToSnippet(
   let targetOpts = config.httpsnippet.targets[target].opts || {};
   const highlightMode = config.highlight;
 
-  // API SDK client needs additional runtime information on the API definition we're showing the
-  // user so it can generate an appropriate snippet.
-  if (language === 'node' && target === 'api') {
-    try {
-      addTargetClient('node', HTTPSnippetSimpleApiClient);
-    } catch (e) {
-      if (!e.message.match(/already exists/)) {
-        throw e;
-      }
-    }
+  plugins.forEach(plugin => {
+    addClientPlugin(plugin);
 
-    targetOpts.apiDefinition = oas ? oas.getDefinition() : null;
-    targetOpts.apiDefinitionUri = opts?.openapi?.registryIdentifier;
-    targetOpts.identifier = opts?.openapi?.variableName;
-  }
+    // Our `httpsnippet-client-api` plugin uses these options so we need to pass them along.
+    if (plugin.target === 'node' && plugin.client.info.key === 'api') {
+      targetOpts.api = {
+        definition: oas ? oas.getDefinition() : null,
+        identifier: opts?.openapi?.variableName,
+        registryURI: opts?.openapi?.registryIdentifier,
+      };
+    }
```

Comment:
maybe i'm misunderstanding how all of this fits together, but i feel like this logic (and its corresponding opts) should be in the plugin code, not here

Reply:
it is, but these options need to get passed along from here into the plugin and the plugin wants these options as an `api` object. im not really sure on the best way to handle a plugin that has its own options like this other than this thing i've got here.